src/OutputHash.js

```diff
@@ -125,34 +131,14 @@ OutputHash.prototype.apply = function apply(compiler) {
         });
 
         compilation.hooks.afterOptimizeAssets.tap('Update chunks', (assets) => {
-            const sortedChunks = [];
-            const visitedGroups = [];
             const nameMap = {};
-
-            const extractInOrder = (group) => {
-                // Mark the group as processed
-                visitedGroups.push(group);
-
-                // For each child group, process it if it hasn't been processed before
-                group.getChildren().forEach((child) => {
-                    if (!visitedGroups.includes(child)) extractInOrder(child);
-                });
-
-                // For each chunk in this group
-                //   - Get all groups containing that chunk (that includes this group)
-                //   - If the group hasn't been processed yet, process it (this will skip current
-                //     group)
-                //   - After all groups containing the chunk have been processed, add the chunk to
-                //     the list of sortedChunks
-                group.chunks.forEach((chunk) => {
-                    Array.from(chunk.groupsIterable).forEach((parentGroup) => {
-                        if (!visitedGroups.includes(parentGroup)) extractInOrder(parentGroup);
-                    });
-                    if (!sortedChunks.includes(chunk)) sortedChunks.push(chunk);
-                });
-            };
-
-            this.chunkGroups.forEach(extractInOrder);
+            const sortedChunks = compilation.chunks.slice().sort(((aChunk, bChunk) => {
```

Comment:
This is the same logic as webpack's [immediately before it performs hashing](https://github.com/webpack/webpack/blob/677eaba1586bfc68adf883161e6d7cca4e0129a9/lib/Compilation.js#L1723-L1729).

Reply:
OMG, so simple. I love it.